### PR TITLE
[CLOUD-3928] Doc and behave tests for custom galleon feature-packs support

### PR DESCRIPTION
**Pull request by jfdenise**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,11 +197,32 @@ file inside your source code repository.
 
       * `web-clustering`: Support for Infinispan-based web session clustering.
 
+      * All layers defined by [wildfly-extras Galleon feature-pack](https://github.com/wildfly-extras/wildfly-datasources-galleon-pack).
```
Comment (jmesnil) [inline README.md:200]: the link text is not correct: wildfly-datasources-feature-pack (this make it explicit that this feature pack deals with datasources)

File: README.md [hunk 2]
```diff
@@ -199,11 +197,32 @@ file inside your source code repository.
...
+
+    * All [WildFly Galleon Layers](https://docs.wildfly.org/23/Galleon_Guide.html#wildfly_layers).
+
+* `GALLEON_PROVISION_FEATURE_PACKS`
+
+     A comma separated list of additional Galleon feature-packs identified by Maven coordinates (`<groupId>:<artifactId>:<version>`). These feature-packs must have been 
+     designed to work with WildFly. Galleon layers defined by these feature-packs can be set in `GALLEON_PROVISION_LAYERS` env variable.
```
Comment (jmesnil) [inline README.md:207]: Should we say that these feature must have been designed to work with the wildlfy-feature-pack that is provisioned .
(what would happen if I specify an extra feature-pack depending on WildFly 22.0 feature-pack to build it against WildFLy 23.0 image?)

File: README.md [hunk 3]
```diff
@@ -304,33 +323,50 @@ Jolokia env variables
...
 
-That is done during s2i build using the `GALLEON_PROVISION_LAYERS` env variable.
+That is done during s2i build by using the `GALLEON_PROVISION_LAYERS` env variable. By default, a fixed set of Galleon layers (WildFly ones and WildFly s2i specific ones) 
+can be used to compose a server. Custom Galleon layers can be incorporated in the provisioned server by setting `GALLEON_PROVISION_FEATURE_PACKS` to reference 
+Galleon feature-packs that provide Galleon layers usable with WildFly.
 
-If you want to define your own WildFly server, create a directory named `galleon` at the root of your application sources project. This directory must
-contains a provisioning.xml file. During s2i build, this file is used to provision a server.
-This server is used to replace the one present in the s2i builder image (located in $JBOSS_HOME).
+Example of a custom `mylayer` Galleon layer included into the provisioned server: `GALLEON_PROVISION_LAYERS=cloud-server,ejb-lite,jpa,mylayer` `GALLEON_PROVISION_FEATURE_PACKS=`org.example:my-galleon-pack:1.0.0.Final`
```
Comment (jmesnil) [inline README.md:336]: maybe reword to specify that the `org.example:my-galleon-pack:1.0.0.Final` that provides the `mylayer` layer.
The relation is not explicit in the existing example

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jfdenise) [reply README.md]: Thanks.

File: README.md (hunk 2 above)
Comment (jfdenise) [reply README.md]: Thanks, good point. I clarified.

File: README.md (hunk 3 above)
Comment (jfdenise) [reply README.md]: Thanks. Fixed.

**Review by jmesnil — APPROVED** [review]

File: test/test-custom-galleon/my/custom/galleon/settings.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<settings xmlns="http://maven.apache.org/settings/1.0.0"
```
Comment (jbliznak) [inline test/test-custom-galleon/my/custom/galleon/settings.xml:1]: Minor and probably would work either way, however, according to https://maven.apache.org/settings.html, the `SETTINGS` should be uppercase
```
    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
```

Comment (jfdenise) [reply test/test-custom-galleon/my/custom/galleon/settings.xml]: @jliznak, Thank-you. Fixed.
